unstructured/ingest/connector/sharepoint.py

```diff
@@ -0,0 +1,312 @@
+from dataclasses import dataclass, field
+from html import unescape
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+from urllib.parse import urlparse
+
+from unstructured.file_utils.filetype import EXT_TO_FILETYPE
+from unstructured.ingest.interfaces import (
+    BaseConnector,
+    BaseConnectorConfig,
+    BaseIngestDoc,
+    ConnectorCleanupMixin,
+    IngestDocCleanupMixin,
+    StandardConnectorConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.utils import requires_dependencies
+
+if TYPE_CHECKING:
+    from office365.sharepoint.files.file import File
```

Comment:
why do we import the File class here but then use the string "File" in the actual typing? can we remove this? benefit that then @requires_dependencies decorators will catch if the library isn't installed